### PR TITLE
Revamps Cyanide

### PR DESCRIPTION
**Pull request by colossusqw**
[LABEL][feature][balance][chemistry]

## About the PR 
This PR remakes cyanide, changing it from dealing 1.5 tox damage per second with some random stuns, to acting in two stages.

Stage 1: this stage comes with barely any tox damage, but emotes and messages. Lasts 25 cycles if the method of application was through the skin, or 13 if it was any other method.

Stage 2: cyanide starts dealing increasing oxygen damage over time, as well as a small amount of heart and lung damage, that whilst not enough to completely kill your heart and lungs, may trigger heart failure or respiratory failure.

It is overall much less stackable, as most of it's danger potential comes after one or two minutes of exposure, but poses a ever increasing threat as long as it is in you. It's progression can be significantly slowed down, however via having space drugs in your system, as it simulates it's interaction with Amyl Nitrate. The speed at which it progresses is also dependant on dose, so 100 units will be roughly twice as fast as 20 units.

## Why's this needed? 
As of now, a lot of poisons are simple tox per second chems, and this poses a problem where stacking can be very effective and kill you much faster than they should. This makes it so small amounts of cyanide wont be anywhere near as strong or stackable, whilst also giving cyanide some extra flavor and separating it a bit from all the other "+2 TOX" chems out there.

cyanide

example of cyanide at work: https://youtu.be/plDGK4McHLY

```changelog
(u)Hans
(*)Revamped Cyanide's behavior.(minor notes for more)
(+)Cyanide now primarily deals oxygen damage, and has a ramp up period before it starts being dangerous, but will become more dangerous with time and cause heart and respiratory failure.
(+)Cyanide's progression can now be significantly slowed down with Space Drugs.
(+)The speed at which cyanide poisoning occurs is now dependant on the dose you are given, for instance 100 units of cyanide will be roughly twice as fast as 20 units.
```

### CODE REVIEW

**Comment by ThatFiveGuys** [comment]
This feels like a big botany hotbox nerf as cyanide is botanys only skinpen chem, I think these changes would be better for a different poison instead of cyanide.

**Comment by colossusqw** [comment]
> This feels like a big botany hotbox nerf as cyanide is botanys only skinpen chem, I think these changes would be better for a different poison instead of cyanide.

considering in this case botany would be hotboxing cyanide by itself and in some considerable amounts, i don't see how this would be much worse than it currently is, it might not be as immediate as the current version, but it's pretty much still a fast acting poison that can mess you up if you don't seek medical attention

**Comment by colossusqw** [comment]
> This seems like a cyanide nerf more than anything, as it's now much more heavily delayed and deals a lot of indirect damage. There's a ton of delayed action poisons out there to the point where another one doesn't really bring much new to the table.
> 
> I'm no dev, but I think it'd be nicer if the progression was slightly faster, and it did some, even if barely any tox damage.

Current cyanide will take about 90 cycles (3 minutes) to kill you, considering the scaling, 10 or 12 units will take about the same, it's mostly that the bulk of the damage isn't instant, though it does hurt you for 0.5 tox per cycle still. This doesn't feel like a nerf to me, expecially with how the damage ramp up goes, just some walking away from instant per cycle damage.

**Comment by TDHooligan** [comment]
this is a nerf. the moment your spaceman does literally anything besides the odd nose pick or cough, you immediately know you have been poisoned and you need to get to medbay. i really don't think chem is in a state that needs this right now

**Comment by colossusqw** [comment]
> this is a nerf. the moment your spaceman does literally anything besides the odd nose pick or cough, you immediately know you have been poisoned and you need to get to medbay. i really don't think chem is in a state that needs this right now

i would definitely not consider this a nerf, the scaling effects make it so it's harder to contain the damage as it progresses, causing cardiac failure and respiratory failure can be big problems as far as treatment goes, and the scaling speed at which it acts should be enough to keep it about where it is at the moment, just more capable on it's own while less usable along with many other poisons

File: code/modules/chemistry/Reagents-PoisonEtc.dm [hunk 1]
```diff
@@ -262,34 +262,56 @@ datum
...
-						if (prob(25))
-							boutput(M, "<span class='alert'>You feel horribly weak.</span>")
-							M.setStatusMin("stunned", 2 SECONDS * mult)
-							M.take_toxin_damage(1.5 * mult)
+				M.take_toxin_damage(0.5 * mult)  // a bit of tox to show you that something is indeed, not alright
+
+				if (holder.get_reagent_amount("space_drugs") > 5) //amyl nitrate inhibits cyanide's effects IRL, space drugs is pretty close
+					counter += 0.25 * mult
+				else
+					counter += max(0.5, (holder.get_reagent_amount(src.id) ** 0.36 / 1.75) * mult)//speed of poisoning scales nonlinearly with dosage
```
Comment (TDHooligan) [inline code/modules/chemistry/Reagents-PoisonEtc.dm:283]: shouldn't this mult be outside of your max? a really slow tick will be capped at +0.5 on the counter

**Comment by TDHooligan** [comment]
>the scaling effects make it so it's harder to contain the damage as it progresses, causing cardiac failure and respiratory failure can be big problems as far as treatment goes, 
after giving it more time to think, i still think it's weaker by virtue of being more easily treatable (heart and respiratory failure are both cured and mostly dealt with by epi) - toxin damage is a lot harder to treat, and cutting more TOX out means easier treatment. 

however, in an ideal scenario where someone doesn't get treated, yeah this one is a bit more reliable come late stage.

**Pull request closed without merging**